[3.4.20] Panic probably due to nil log object

Based on the feedback from @JohnJAS, this issue can't be reproduced in 3.4.19, instead it can only be reproduced on 3.4.20.

It looks like the [lg](https://github.com/etcd-io/etcd/blob/1e2682301c2947619214e6497b5432e7717d921d/etcdserver/v3_server.go#L805) is `nil`. 

Thanks both @JohnJAS and @rtheis. 

```
2022-08-29 06:29:02.702707 E | rafthttp: failed to read 88d5d8ebd854df5a on stream MsgApp v2 (unexpected EOF)
2022-08-29 06:29:02.702723 I | rafthttp: peer 88d5d8ebd854df5a became inactive (message send to peer failed)
2022-08-29 06:29:02.703504 W | rafthttp: lost the TCP streaming connection with peer 88d5d8ebd854df5a (stream Message reader)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x75745b]

goroutine 287 [running]:
go.uber.org/zap.(*Logger).check(0x0, 0x1, 0x10a3ea5, 0x36, 0xc003fb7480)
        /root/go/pkg/mod/go.uber.org/zap@v1.10.0/logger.go:264 +0x9b
go.uber.org/zap.(*Logger).Warn(0x0, 0x10a3ea5, 0x36, 0xc003fb7480, 0x2, 0x2)
        /root/go/pkg/mod/go.uber.org/zap@v1.10.0/logger.go:194 +0x45
go.etcd.io/etcd/etcdserver.(*EtcdServer).requestCurrentIndex(0xc000398600, 0xc005210f00, 0xd15682e845891c12, 0x0, 0x0, 0x0)
        /tmp/etcd-release-3.4.20/etcd/release/etcd/etcdserver/v3_server.go:805 +0x873
go.etcd.io/etcd/etcdserver.(*EtcdServer).linearizableReadLoop(0xc000398600)
        /tmp/etcd-release-3.4.20/etcd/release/etcd/etcdserver/v3_server.go:721 +0x2d6
go.etcd.io/etcd/etcdserver.(*EtcdServer).goAttach.func1(0xc000398600, 0xc00011ec20)
        /tmp/etcd-release-3.4.20/etcd/release/etcd/etcdserver/server.go:2698 +0x57
created by go.etcd.io/etcd/etcdserver.(*EtcdServer).goAttach
        /tmp/etcd-release-3.4.20/etcd/release/etcd/etcdserver/server.go:2696 +0x1b1
```

--- a/etcdserver/v3_server.go
+++ b/etcdserver/v3_server.go
@@ -779,11 +779,15 @@ func (s *EtcdServer) requestCurrentIndex(leaderChangedNotifier <-chan struct{}, 
 				if len(rs.RequestCtx) == 8 {
 					responseId = binary.BigEndian.Uint64(rs.RequestCtx)
 				}
-				lg.Warn(
-					"ignored out-of-date read index response; local node read indexes queueing up and waiting to be in sync with leader",
-					zap.Uint64("sent-request-id", requestId),
-					zap.Uint64("received-request-id", responseId),
-				)
+				if lg != nil {
+					lg.Warn(
+						"ignored out-of-date read index response; local node read indexes queueing up and waiting to be in sync with leader",
+						zap.Uint64("sent-request-id", requestId),
+						zap.Uint64("received-request-id", responseId),
+					)
+				} else {
+					plog.Warningf("ignored out-of-date read index response; local node read indexes queueing up and waiting to be in sync with leader (request ID want %d, got %d)", requestId, responseId)
+				}
 				slowReadIndex.Inc()
 				continue
 			}
@@ -794,7 +798,11 @@ func (s *EtcdServer) requestCurrentIndex(leaderChangedNotifier <-chan struct{}, 
 			return 0, ErrLeaderChanged
 		case <-firstCommitInTermNotifier:
 			firstCommitInTermNotifier = s.FirstCommitInTermNotify()
-			lg.Info("first commit in current term: resending ReadIndex request")
+			if lg != nil {
+				lg.Info("first commit in current term: resending ReadIndex request")
+			} else {
+				plog.Info("first commit in current term: resending ReadIndex request")
+			}
 			err := s.sendReadIndex(requestId)
 			if err != nil {
 				return 0, err
@@ -802,11 +810,15 @@ func (s *EtcdServer) requestCurrentIndex(leaderChangedNotifier <-chan struct{}, 
 			retryTimer.Reset(readIndexRetryTime)
 			continue
 		case <-retryTimer.C:
-			lg.Warn(
-				"waiting for ReadIndex response took too long, retrying",
-				zap.Uint64("sent-request-id", requestId),
-				zap.Duration("retry-timeout", readIndexRetryTime),
-			)
+			if lg != nil {
+				lg.Warn(
+					"waiting for ReadIndex response took too long, retrying",
+					zap.Uint64("sent-request-id", requestId),
+					zap.Duration("retry-timeout", readIndexRetryTime),
+				)
+			} else {
+				plog.Warningf("waiting for ReadIndex response took too long, retrying (sent-request-id: %d, retry-timeout: %s)", requestId, readIndexRetryTime)
+			}
 			err := s.sendReadIndex(requestId)
 			if err != nil {
 				return 0, err
@@ -814,10 +826,14 @@ func (s *EtcdServer) requestCurrentIndex(leaderChangedNotifier <-chan struct{}, 
 			retryTimer.Reset(readIndexRetryTime)
 			continue
 		case <-errorTimer.C:
-			lg.Warn(
-				"timed out waiting for read index response (local node might have slow network)",
-				zap.Duration("timeout", s.Cfg.ReqTimeout()),
-			)
+			if lg != nil {
+				lg.Warn(
+					"timed out waiting for read index response (local node might have slow network)",
+					zap.Duration("timeout", s.Cfg.ReqTimeout()),
+				)
+			} else {
+				plog.Warningf("timed out waiting for read index response (local node might have slow network) timeout: %s", s.Cfg.ReqTimeout())
+			}
 			slowReadIndex.Inc()
 			return 0, ErrTimeout
 		case <-s.stopping:

